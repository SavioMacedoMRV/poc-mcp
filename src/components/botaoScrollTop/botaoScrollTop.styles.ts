import styled from 'styled-components'
import {margens} from 'resources/margens'
import {cores} from 'resources/cores'
import SetaSVG from 'icons/iSeta.svg'
import {Icone} from 'components/icone'
import {Texto} from 'components/texto'

export const BotaoStyled = styled.button<{$exibir?: boolean}>`
  border: none;
  display: ${({$exibir}) => ($exibir ? 'flex' : 'none')};
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-bottom: ${margens.medium}px;
  background: transparent;

  cursor: pointer;
  &:hover {
    opacity: 0.7;
  }
  &:active {
    opacity: 1;
  }
`

export const IconeScrollTop = styled(Icone).attrs({
  icone: SetaSVG,
  cor: cores.neutralMedium,
  altura: 48,
  inclinacao: -90,
})``

export const TextoScrollTop = styled(Texto).attrs({
  tamanho: 24,
  alturalinha: 30,
  cursor: 'pointer',
  estilo: 'semibold',
  cor: cores.neutralMedium,
})``
