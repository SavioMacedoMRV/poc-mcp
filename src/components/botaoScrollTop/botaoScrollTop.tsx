import React, {useEffect, useState} from 'react'
import * as S from './botaoScrollTop.styles'
import {Evento} from 'types/enums'
import {RegistrarEvento} from 'infrastructure/logs'

interface BotaoScrollTopProps {
  aoClicar?: () => void
}

export const BotaoScrollTop = ({aoClicar}: Readonly<BotaoScrollTopProps>) => {
  const [exibirBotaoScrollTop, setExibirBotaoScrollTop] = useState(false)

  const aoClicarScrollTop = () => {
    RegistrarEvento({evento: Evento.BotaoScrollTopClick})

    window.scrollTo({
      top: 0,
      behavior: 'smooth',
    })

    if (aoClicar) {
      aoClicar()
    }
  }

  useEffect(() => {
    const handleScroll = () => {
      const scrollPosition = window.scrollY
      const windowHeight = window.innerHeight

      setExibirBotaoScrollTop(scrollPosition > windowHeight / 4)
    }

    window.addEventListener('scroll', handleScroll)

    return () => {
      window.removeEventListener('scroll', handleScroll)
    }
  }, [])

  return (
    <S.BotaoStyled
      data-testid={'btn-scroll-top'}
      $exibir={exibirBotaoScrollTop}
      onClick={aoClicarScrollTop}
    >
      <S.IconeScrollTop />
      <S.TextoScrollTop>voltar ao in√≠cio</S.TextoScrollTop>
    </S.BotaoStyled>
  )
}
