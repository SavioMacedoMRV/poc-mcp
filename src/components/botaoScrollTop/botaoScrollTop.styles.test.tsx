import {render} from '@testing-library/react'
import * as S from './botaoScrollTop.styles'

describe('BotaoScrollTop.styles', () => {
  test('BotaoStyled DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.BotaoStyled />)
    expect(container).toMatchSnapshot()
  })

  test('BotaoStyled DEVE ser igual ao snapshot QUANDO $exibir for verdadeiro', () => {
    const {container} = render(<S.BotaoStyled $exibir />)
    expect(container).toMatchSnapshot()
  })

  test('IconeScrollTop DEVE ser igual ao snapshot', () => {
    const container = render(<S.IconeScrollTop />)
    expect(container).toMatchSnapshot()
  })

  test('TextoScrollTop DEVE ser igual ao snapshot', () => {
    const container = render(<S.TextoScrollTop />)
    expect(container).toMatchSnapshot()
  })
})
