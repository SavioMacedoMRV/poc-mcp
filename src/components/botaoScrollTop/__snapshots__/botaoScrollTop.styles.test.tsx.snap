// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BotaoScrollTop.styles BotaoStyled DEVE ser igual ao snapshot 1`] = `
.c0 {
  border: none;
  display: none;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-bottom: 24px;
  background: transparent;
  cursor: pointer;
}

.c0:hover {
  opacity: 0.7;
}

.c0:active {
  opacity: 1;
}

<div>
  <button
    class="c0"
  />
</div>
`;

exports[`BotaoScrollTop.styles BotaoStyled DEVE ser igual ao snapshot QUANDO $exibir for verdadeiro 1`] = `
.c0 {
  border: none;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-bottom: 24px;
  background: transparent;
  cursor: pointer;
}

.c0:hover {
  opacity: 0.7;
}

.c0:active {
  opacity: 1;
}

<div>
  <button
    class="c0"
  />
</div>
`;

exports[`BotaoScrollTop.styles IconeScrollTop DEVE ser igual ao snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        class=""
        data-file-name="SvgISeta"
        fill="#999E9C"
        height="48"
        style="min-width: 48px; min-height: 48px; max-width: 48px; max-height: 48px; transform: rotate(-90deg);"
        width="48"
      />
    </div>
  </body>,
  "container": <div>
    <svg
      class=""
      data-file-name="SvgISeta"
      fill="#999E9C"
      height="48"
      style="min-width: 48px; min-height: 48px; max-width: 48px; max-height: 48px; transform: rotate(-90deg);"
      width="48"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BotaoScrollTop.styles TextoScrollTop DEVE ser igual ao snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  font-weight: 400;
  font-size: 24px;
  line-height: 30px;
  font-family: AvertaStd-Semibold;
  color: #999E9C;
  text-align: start;
  cursor: pointer;
  text-transform: none;
}

<body>
    <div>
      <span
        alturalinha="30"
        class="c0 "
        cor="#999E9C"
        cursor="pointer"
        estilo="semibold"
        tamanho="24"
      />
    </div>
  </body>,
  "container": .c0 {
  font-weight: 400;
  font-size: 24px;
  line-height: 30px;
  font-family: AvertaStd-Semibold;
  color: #999E9C;
  text-align: start;
  cursor: pointer;
  text-transform: none;
}

<div>
    <span
      alturalinha="30"
      class="c0 "
      cor="#999E9C"
      cursor="pointer"
      estilo="semibold"
      tamanho="24"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
